### PR TITLE
DATAUP-500 Add messages to enable/disable options in select input

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

This alters the selectInput widget in the following ways
1. Adds an optional `availableValues` config option, which, if present, overrides the values taken from the constant spec.
2. Adds an optional `disabledValues` config, which disables any of the set values on startup.
3. Adds the `disable-values` and `enable-values` messages, which each take an Array of value strings. If any are present in the dropdown they'll be disabled/enabled. If any of those are the currently selected value, it stays enabled until the user selects a separate option. It doesn't seem to make sense to disable the selected value.

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-500
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
* Nothing really obviously user-facing yet - commands will be included in the next PR.
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/common/cellComponents/fieldTableCellWidget.js [hunk 1]
```diff
@@ -22,19 +22,8 @@ define([
         let places, parent, container, inputControl;
 
         try {
-            inputControl = inputControlFactory.make({
-                bus: bus,
-                paramsChannelName: config.paramsChannelName,
-                channelName: bus.channelName,
-                initialValue: config.initialValue,
```
Comment (briehl) [inline kbase-extension/static/kbase/js/common/cellComponents/fieldTableCellWidget.js:29]: Most of this is copy-pasted from the config object, so I just extended them.
This also allows any input-widget-specific things (like `availableValues` and `disabledValues` in this case) to just freely pass in.

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/selectInput.js [hunk 2]
```diff
@@ -19,16 +19,34 @@ define([
...
     function factory(config) {
         const spec = config.parameterSpec,
             runtime = Runtime.make(),
             busConnection = runtime.bus().connect(),
             channel = busConnection.channel(config.channelName),
             model = {
-                availableValues: spec.data.constraints.options,
+                availableValues: config.availableValues || spec.data.constraints.options,
                 value: config.initialValue,
+                disabledValues: new Set(config.disabledValues || []),
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/selectInput.js:46]: Making these `Sets` so they're faster lookup times (assuming lots of file inputs), and pretty broadly browser-friendly at this point (I don't think there's a Set polyfill set up) - https://caniuse.com/?search=new%20Set

Comment (briehl) [reply kbase-extension/static/kbase/js/widgets/appWidgets2/input/selectInput.js:46]: If it's a problem, the [](MDN Set Page) recommends [https://github.com/zloirock/core-js#set](this polyfill)

Comment (ialarmedalien) [reply kbase-extension/static/kbase/js/widgets/appWidgets2/input/selectInput.js:46]: I'm fairly sure we have sets elsewhere anyway.

**Review by ialarmedalien — APPROVED** [review]
👍

**Pull request merged**